python-lib/custom_gluon_models/arima.py

```diff
@@ -0,0 +1,134 @@
+from gluonts.model.estimator import Estimator
+from gluonts.model.forecast import SampleForecast
+from gluonts.model.predictor import RepresentablePredictor
+from gluonts.support.pandas import frequency_add
+from gluonts.core.component import validated
+from gluonts.time_feature import get_seasonality
+from custom_gluon_models.utils import cast_kwargs
+from constants import TIMESERIES_KEYS
+import pmdarima as pm
+import numpy as np
+from safe_logger import SafeLogger
+from tqdm import tqdm
+
+
+logger = SafeLogger("Forecast plugin - ARIMA")
+
+
+class ArimaPredictor(RepresentablePredictor):
+    """
+    An abstract predictor that can be subclassed by models that are not based
+    on Gluon. Subclasses should have @validated() constructors.
+    (De)serialization and value equality are all implemented on top of the
+    @validated() logic.
+
+    Parameters
+    ----------
+    prediction_length
+        Prediction horizon.
+    freq
+        Frequency of the predicted data.
+    """
+
+    # TODO implement custom serializer
+
+    @validated()
+    def __init__(self, prediction_length, freq, trained_models, lead_time=0):
+        super().__init__(freq=freq, lead_time=lead_time, prediction_length=prediction_length)
+        self.trained_models = trained_models
+
+    def predict(self, dataset, **kwargs):
+        """
+
+        Args:
+            dataset (gluonts.dataset.common.Dataset): Dataset after wich to predict forecasts.
+
+        Yields:
+            SampleForecast of predictions.
+        """
+        logger.info("Prediction timeseries ...")
```

Comment:
Could the logger be a bit more specific?

Reply:
545a70f
and the logger is initialised with "Auto ARIMA"